src/main/java/uk/gov/hmcts/reform/slc/services/ReportParser.java

```diff
@@ -0,0 +1,67 @@
+package uk.gov.hmcts.reform.slc.services;
+
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.slc.model.LetterPrintStatus;
+import uk.gov.hmcts.reform.slc.services.steps.getpdf.FileNameHelper;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.time.ZonedDateTime;
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.StreamSupport.stream;
+
+@Component
+public class ReportParser {
+
+    private static final Logger logger = LoggerFactory.getLogger(ReportParser.class);
+
+    public List<LetterPrintStatus> parse(byte[] report) {
+        try (CSVParser parser = parserFor(report)) {
+
+            return stream(parser.spliterator(), false)
+                .map(row -> toPrintStatus(row))
+                .filter(status -> status != null)
+                .collect(toList());
+
+        } catch (IOException exc) {
+            throw new ReportParsingException(exc);
+        }
+    }
+
+    private CSVParser parserFor(byte[] csv) throws IOException {
+        return CSVFormat
+            .DEFAULT
+            .withHeader()
+            .parse(new InputStreamReader(new ByteArrayInputStream(csv)));
+    }
+
+    /**
+     * Converts cvs row into a letter print status object.
+     * Returns null if conversion fails.
+     */
+    private LetterPrintStatus toPrintStatus(CSVRecord record) {
+        try {
+            return new LetterPrintStatus(
+                FileNameHelper.extractId(record.get("Filename")),
+                ZonedDateTime.parse(record.get("Date") + "T" + record.get("Time") + "Z")
+            );
+        } catch (Exception exc) {
+            logger.error("Error parsing row: " + record, exc);
```

Comment:
Perhaps even some metrics/events to app insights. Probably not for this pull request.